diff -up index-fm-v2.1.0/CMakeLists.txt.omv~ index-fm-v2.1.0/CMakeLists.txt
diff -up index-fm-v2.1.0/src/CMakeLists.txt.omv~ index-fm-v2.1.0/src/CMakeLists.txt
--- index-fm-v2.1.0/src/CMakeLists.txt.omv~	2021-12-29 20:59:59.215615972 +0100
+++ index-fm-v2.1.0/src/CMakeLists.txt	2021-12-29 21:01:18.998775934 +0100
@@ -39,7 +39,7 @@ if (ANDROID)
     #    set_target_properties(create-apk-${PROJECT_NAME} PROPERTIES ANDROID_APK_DIR "${APK_DIR}")
 
 elseif(UNIX)
-    target_link_libraries(${PROJECT_NAME} KF5::KIOCore)
+    target_link_libraries(${PROJECT_NAME} KF5::KIOCore KF5::KIOGui)
 endif()
 
 if(UNIX OR WIN32 OR APPLE AND NOT ANDROID)
diff -up index-fm-v2.1.0/src/index.cpp.omv~ index-fm-v2.1.0/src/index.cpp
--- index-fm-v2.1.0/src/index.cpp.omv~	2021-12-29 20:58:27.779457079 +0100
+++ index-fm-v2.1.0/src/index.cpp	2021-12-29 21:02:46.622970954 +0100
@@ -6,7 +6,7 @@
 #include "index.h"
 
 #if defined Q_OS_LINUX && !defined Q_OS_ANDROID
-#include <KToolInvocation>
+#include <KTerminalLauncherJob>
 #endif
 
 #include <QDebug>
@@ -44,9 +44,9 @@ void Index::openPaths(const QStringList
 void Index::openTerminal(const QUrl &url)
 {
 #if defined Q_OS_LINUX && !defined Q_OS_ANDROID
-
-    KToolInvocation::invokeTerminal(QString(), {}, url.isLocalFile() ? url.toLocalFile() : FMStatic::HomePath);
-
+    auto job = new KTerminalLauncherJob(QString());
+    job->setWorkingDirectory(url.isLocalFile() ? url.toLocalFile() : FMStatic::HomePath);
+    job->start();
 #else
     Q_UNUSED(url)
 #endif
